### ISSUE TITLE
Can't pip install PyTorch 1.4.0 with CUDA 10.1

### ISSUE THREAD

**Issue by VivekPanyam** [pytorch/pytorch#37113]
## 🐛 Bug

The default CUDA version for the PyTorch 1.4.0 wheels is CUDA 10.1, but `pip install` defaults to CUDA 9.2 and does not let you install the 10.1 wheel.

Across Python 2.7, 3.6, 3.7, 3.8:

`pip install -f https://download.pytorch.org/whl/torch_stable.html torch==1.4.0`

will install the `cu92` wheel instead of the `cu101` wheel.

The following command complains that it can't find the requested version
`pip install -f https://download.pytorch.org/whl/torch_stable.html torch==1.4.0+cu101`

This makes sense as the 10.1 wheels for torch 1.4.0 don't have a suffix. Maybe this is related to the 1.5.0 release from yesterday?

## To Reproduce

1. `docker run -it --rm python:2.7 /bin/bash` (replacing `2.7` with each python version above)
2. `pip install -f https://download.pytorch.org/whl/torch_stable.html torch==1.4.0`

```
root@40e0b8aa1d83:/# pip install -f https://download.pytorch.org/whl/torch_stable.html torch==1.4.0
Looking in links: https://download.pytorch.org/whl/torch_stable.html
Collecting torch==1.4.0
  Downloading https://download.pytorch.org/whl/cu92/torch-1.4.0%2Bcu92-cp38-cp38-linux_x86_64.whl (640.6 MB)
     |▌                               | 10.3 MB 9.9 MB/s eta 0:01:04^C
ERROR: Operation cancelled by user
root@40e0b8aa1d83:/#
```

3. `pip install -f https://download.pytorch.org/whl/torch_stable.html torch==1.4.0+cu101`

```
root@40e0b8aa1d83:/# pip install -f https://download.pytorch.org/whl/torch_stable.html torch==1.4.0+cu101
Looking in links: https://download.pytorch.org/whl/torch_stable.html
ERROR: Could not find a version that satisfies the requirement torch==1.4.0+cu101 (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2, 1.4.0, 1.4.0+cpu, 1.4.0+cu100, 1.4.0+cu92, 1.5.0, 1.5.0+cpu, 1.5.0+cu101, 1.5.0+cu92)
ERROR: No matching distribution found for torch==1.4.0+cu101
root@40e0b8aa1d83:/#
```

## Expected behavior

`pip install -f https://download.pytorch.org/whl/torch_stable.html torch==1.4.0`

should install Torch 1.4.0 with CUDA 10.1 support

**Comment by YuanYingdong** [comment]
Maybe you can try 1.5 with command 
```
pip install torch==1.5.0+cu101 torchvision==0.6.0+cu101 -f https://download.pytorch.org/whl/torch_stable.html
```

**Comment by VivekPanyam** [comment]
Yes, installing 1.5.0 does work, but this is specifically about 1.4.0 + cuda 10.1 failing

A workaround is to install 1.4.0 + CUDA 10.1 by using the wheel path directly (e.g. `pip install https://download.pytorch.org/whl/cu101/torch-1.4.0-cp38-cp38-linux_x86_64.whl`), but this isn't ideal

**Comment by peterjc123** [comment]
It can be fixed by using three equal signs instead of two.

**Comment by mruberry** [comment]
Thank you for reporting this issue, @VivekPanyam! Glad to hear you have a workaround, too.

**Comment by vraimonds** [comment]
Having the same issue here!

Both workarounds worked for me. The easiest:

> It can be fixed by using three equal signs instead of two.

For the other one just need to not forget to find the right wheel from https://download.pytorch.org/whl/torch_stable.html

> Yes, installing 1.5.0 does work, but this issue is specifically about 1.4.0 + cuda 10.1 failing
> 
> A workaround is to install 1.4.0 + CUDA 10.1 by using the wheel path directly (e.g. `pip install https://download.pytorch.org/whl/cu101/torch-1.4.0-cp38-cp38-linux_x86_64.whl`), but this isn't ideal

**Comment by tomzx** [comment]
This problem applies to the following:
* torch 1.4.0, CUDA 10.1
* torch 1.5.1, CUDA 10.2
* torch 1.6.0, CUDA 10.2

It appears that the whl files in these cases do not contain `+cu$VERSION` as part of their name. Is this intended to indicate those are not supported but can still be installed if linked to directly?
